### DISCUSSION TITLE
Should official install-included plugins be available through `$env.NU_PLUGIN_DIRS`?

### DISCUSSION THREAD

**Discussion by Kissaki** [nushell/nushell#16658] [Q&A]
### Question

[`env.nu` sets up `$env.NU_PLUGIN_DIRS`](https://github.com/nushell/nushell/blob/a3bc85bb5fba0d29f5243e5451cd8088b84af0c8/crates/nu-utils/src/sample_config/default_env.nu#L84-L86) with as an array of `($nu.default-config-dir | path join 'plugins')`.

The Windows installer installs four official plugins inside `nu\bin\`: `nu_plugin_gstat.exe`, `nu_plugin_inc.exe`, `nu_plugin_polars.exe`, `nu_plugin_query.exe`

The `README.md` next to it says you can add them with `plugin add ./nu_plugin_query` - a local path.

I would expect the official plugins to be addable without knowing about, locating, and running qualified path commands to activate them.

I would expect `plugin add nu_plugin_query` to work out of the box.

Is there behavior or context I am missing and that is not documented? Does nu fall back to the bin folder?
Would it make more sense to have those plugins in a dedicated plugin folder that is included in `$env.NU_PLUGIN_DIRS`?

### Additional context and details

I am not currently able to add them like that, which may or may not be related to #13040.

The [Plugins doc](https://www.nushell.sh/book/plugins.html) does not talk about this.

**Comment by Kissaki** [comment]
Fixup in `env.nu`:

```nu
$env.NU_PLUGIN_DIRS = ($env.NU_PLUGIN_DIRS | prepend ($nu.current-exe | path parse | get parent))
```

---

<details><summary>plugin add for all core plugins…</summary>

```nu
ls ($nu.current-exe | path parse | get parent)
plugin add nu_plugin_gstat.exe
plugin add nu_plugin_inc.exe
plugin add nu_plugin_polars.exe
plugin add nu_plugin_query.exe
plugin list
```

</details>

**Comment by WindSoilder** [comment]
Closing the issue because the question seems resolved

**Comment by Kissaki** [comment]
Resolved how?

**Comment by WindSoilder** [comment]
Oh, I see you have fixed up in your comment, then I think it's resolved.
I will reopen it if you think it's necessary.  Sorry for the noise

**Comment by Kissaki** [comment]
It's a manual workaround.

**Comment by weirdan** [comment]
Note that plugin paths differ between platforms. For example, the official nushell DEB distribution puts bundled plugins into `/usr/libexec/nushell`, and this folder is not in `$NU_PLUGIN_DIRS` by default.
